# # Mongodb의 categoryinfo 컬렉션을 이용해서, 카테고리 매핑 태이블을 살펴보자

In [3]:
import pandas as pd

### # categoryinfo 컬렉션 :  categoryinfo_.xlsx

In [4]:
cateinfo = pd.read_excel('./categoryinfo_.xlsx')

In [5]:
cateinfo.head()

_id  dcate_cd     dcate_nm     id1       id2  id3  id4  id5  \
0  100602_4100005_L   4100005    초코파이/카스타드  100602  100600.0    0    0    0   
1  100001_4100019_L   4100019  케이크/조각케익/파이  100001  100000.0    0    0    0   
2  100001_4100026_L   4100026      초콜릿/초코볼  100001  100000.0    0    0    0   
3  100001_4100030_L   4100030     냉면/쫄면/소바  100001  100000.0    0    0    0   
4  100621_4100030_L   4100030     냉면/쫄면/소바  100621  100600.0    0    0    0   

   lcate_cd lcate_nm     ...       mcate_nm         name1         name2 name3  \
0   1100000     가공식품     ...      과자/베이커리/떡         과자/간식            식품   NaN   
1   1100000     가공식품     ...      과자/베이커리/떡  추천^식품/생활/유아동  추천^식품/생활/유아동   NaN   
2   1100000     가공식품     ...      과자/베이커리/떡  추천^식품/생활/유아동  추천^식품/생활/유아동   NaN   
3   1100000     가공식품     ...          라면/면류  추천^식품/생활/유아동  추천^식품/생활/유아동   NaN   
4   1100000     가공식품     ...          라면/면류         라면/면류            식품   NaN   

  name4 name5   scate_cd   scate_nm type Unnamed: 20  
0   NaN   NaN  3100000.0  과자/안주/스낵류    L         NaN  
1   NaN   NaN  3100002.0     빵/베이커리    L         NaN  
2   NaN   NaN  3100004.0   초콜릿/사탕/껌    L         NaN  
3   NaN   NaN  3100005.0       국수/면    L         NaN  
4   NaN   NaN  3100005.0       국수/면    L         NaN  

[5 rows x 21 columns]

## # 카테고리 종류 확인

In [6]:
cateinfo[cateinfo.type == "L"].shape # GNB 카테고리 매핑

(16434, 21)

In [7]:
cateinfo[cateinfo.type == "T"].shape # 물리 카테고리 매핑

(9044, 21)

: 라이브딜 / 배송2.0 카테고리 규모 - 1:5 정도였다.

## # 이후에는 물리 카테고리 매핑만 살펴보도록 하자 :  cate_

In [8]:
# 물리 카테고리 매핑만 살펴보자

cate_ = cateinfo[cateinfo.type == "T"]

In [9]:
cate_.shape

(9044, 21)

---

## # cate_에서의 id1~id4는 표준 카테고리의 ti1~ti4를 의미한다.

In [10]:
cate_[cate_.id1 == 232010112] # 232010112 카테고리에 매핑된 배송2.0 카테고리

_id  dcate_cd       dcate_nm        id1          id2  \
3705   232010112_4101094_T   4101094  이외 유아/어린이 DVD  232010112  232010100.0   
13685  232010112_4101092_T   4101092            동요집  232010112  232010100.0   
13768  232010112_4105313_T   4105313         유아동DVD  232010112  232010100.0   
18708  232010112_4101095_T   4101095      해외/영어 DVD  232010112  232010100.0   
23635  232010112_4101093_T   4101093     애니메이션/교육만화  232010112  232010100.0   
23717  232010112_4105238_T   4105238         유아동DVD  232010112  232010100.0   

             id3        id4        id5  lcate_cd  lcate_nm     ...      \
3705   232010000  232000000  200000000   1100005  도서/교육/음반     ...       
13685  232010000  232000000  200000000   1100005  도서/교육/음반     ...       
13768  232010000  232000000  200000000   1100005  도서/교육/음반     ...       
18708  232010000  232000000  200000000   1100005  도서/교육/음반     ...       
23635  232010000  232000000  200000000   1100005  도서/교육/음반     ...       
23717  232010000  232000000  200000000   1100005  도서/교육/음반     ...       

        mcate_nm   name1 name2   name3  name4  name5   scate_cd      scate_nm  \
3705   유아동/청소년도서  유아동/교육   DVD  DVD/음반  도서/교육  배송상품군  3100206.0    유아/어린이 DVD   
13685  유아동/청소년도서  유아동/교육   DVD  DVD/음반  도서/교육  배송상품군  3100206.0    유아/어린이 DVD   
13768  중고 DVD/음반  유아동/교육   DVD  DVD/음반  도서/교육  배송상품군  3100928.0  DVD/비디오/블루레이   
18708  유아동/청소년도서  유아동/교육   DVD  DVD/음반  도서/교육  배송상품군  3100206.0    유아/어린이 DVD   
23635  유아동/청소년도서  유아동/교육   DVD  DVD/음반  도서/교육  배송상품군  3100206.0    유아/어린이 DVD   
23717     DVD/음반  유아동/교육   DVD  DVD/음반  도서/교육  배송상품군  3100184.0  DVD/비디오/블루레이   

      type Unnamed: 20  
3705     T         NaN  
13685    T         NaN  
13768    T         NaN  
18708    T         NaN  
23635    T         NaN  
23717    T         NaN  

[6 rows x 21 columns]

## # 노트북 파일, <u>"카테고리 매핑 데이터 핸들링_ _1"</u>을 통해 정제된 데이터 <u>'df2__re'</u> 불러오자 : 

노트북 : normal

_노트북_ : under bar 1개

__노트북__ : under bar 2개

___노트북___ : under bar 3개

In [11]:

df2_re = pd.read_csv('df2_re.csv')

In [12]:
df2_re.shape

(1542, 8)

In [13]:
df2_re.head()

Unnamed: 0  index    dcateCd  rcp_diff     rcp_x     rcp_y  \
0           0     98  232010112  0.548803  0.118132  0.666935   
1           1    568  233090606  0.432881  0.040000  0.472881   
2           2    124  232040102  0.371951  0.000000  0.371951   
3           3    146  232050207  0.349552  0.014085  0.363636   
4           4    564  233090602  0.340484  0.253012  0.593496   

   totalOrderUserCnt_x  totalOrderUserCnt_y  
0                728.0                 2486  
1                 50.0                  590  
2                 24.0                  164  
3                 71.0                  330  
4                166.0                  984

---

## (***출발!)# 표준카테고리에, 어떠한 물리 카테고리 정보들이 mapping 되었는지 알고싶다!

### # 연습

In [16]:
df2_re.dcateCd[0]

232010112

In [20]:
c = {}

In [23]:
c[0] = cate_[cate_.id1 == 232010112].dcate_cd.values

In [24]:
c[0]

array([4101094, 4101092, 4105313, 4101095, 4101093, 4105238])

In [25]:
d = {}

In [26]:
d[0] = cate_[cate_.id1 == 232010112].dcate_nm.values

In [27]:
d[0]

array(['이외 유아/어린이 DVD', '동요집', '유아동DVD', '해외/영어 DVD', '애니메이션/교육만화',
       '유아동DVD'], dtype=object)

## # 카테고리 정보들을 mapping 해보자 : 단, df2_re 데이터를 활용해서!

In [18]:
df2_re.dcateCd[0]

232010112

---

In [21]:
cate_.columns

Index(['_id', 'dcate_cd', 'dcate_nm', 'id1', 'id2', 'id3', 'id4', 'id5',
       'lcate_cd', 'lcate_nm', 'mcate_cd', 'mcate_nm', 'name1', 'name2',
       'name3', 'name4', 'name5', 'scate_cd', 'scate_nm', 'type',
       'Unnamed: 20'],
      dtype='object')

#### # category_info 컬렉션에서, 표준 카테고리 매핑 & '배송 카테고리' 만 살펴보자!

In [76]:
# category_info 컬렉션에서, 표준 카테고리 매핑 & '배송 카테고리' 만 살펴보자!

cate_deli = cate_[(200000000 <= cate_.id1) & (cate_.id1 < 300000000)]

#### # 물리카테고리를 표준카테고리로 mapping한것이니깐, 물리카테고리의 개수가 데이터의 '행'과 같아야 하는게 맞지 않나?

In [24]:
cate_deli.shape

(6954, 21)

In [35]:
cate_deli.dcate_cd.nunique() # 1100개 정도가 빠지네..?

5831

In [78]:
cate_deli[cate_deli.dcate_cd==0].shape

(1124, 21)

In [32]:
cate_deli[cate_deli.dcate_cd.duplicated()]

_id  dcate_cd dcate_nm        id1          id2        id3  \
4195   231010603_0_T         0      NaN  231010603  231010600.0  231010000   
4196   231020000_0_T         0      NaN  231020000  231000000.0  200000000   
4197   231030200_0_T         0      NaN  231030200  231030000.0  231000000   
4198   231030201_0_T         0      NaN  231030201  231030200.0  231030000   
4199   231040300_0_T         0      NaN  231040300  231040000.0  231000000   
4200   231060100_0_T         0      NaN  231060100  231060000.0  231000000   
4201   231060201_0_T         0      NaN  231060201  231060200.0  231060000   
4202   232000000_0_T         0      NaN  232000000  200000000.0          0   
4203   232010113_0_T         0      NaN  232010113  232010100.0  232010000   
4204   232020000_0_T         0      NaN  232020000  232000000.0  200000000   
4205   232020100_0_T         0      NaN  232020100  232020000.0  232000000   
4206   232030000_0_T         0      NaN  232030000  232000000.0  200000000   
4207   232030100_0_T         0      NaN  232030100  232030000.0  232000000   
4208   232030103_0_T         0      NaN  232030103  232030100.0  232030000   
4209   232030302_0_T         0      NaN  232030302  232030300.0  232030000   
4210   232030304_0_T         0      NaN  232030304  232030300.0  232030000   
4211   232030505_0_T         0      NaN  232030505  232030500.0  232030000   
4212   232030603_0_T         0      NaN  232030603  232030600.0  232030000   
4213   232030604_0_T         0      NaN  232030604  232030600.0  232030000   
4214   232030700_0_T         0      NaN  232030700  232030000.0  232000000   
4215   232040200_0_T         0      NaN  232040200  232040000.0  232000000   
4216   232050100_0_T         0      NaN  232050100  232050000.0  232000000   
4217   232050104_0_T         0      NaN  232050104  232050100.0  232050000   
4218   232050302_0_T         0      NaN  232050302  232050300.0  232050000   
4219   232050600_0_T         0      NaN  232050600  232050000.0  232000000   
4220   232050902_0_T         0      NaN  232050902  232050900.0  232050000   
4221   232050903_0_T         0      NaN  232050903  232050900.0  232050000   
4222   232051000_0_T         0      NaN  232051000  232050000.0  232000000   
4223   232060000_0_T         0      NaN  232060000  232000000.0  200000000   
4224   232060100_0_T         0      NaN  232060100  232060000.0  232000000   
...              ...       ...      ...        ...          ...        ...   
25269  242000000_0_T         0      NaN  242000000  200000000.0          0   
25270  242020000_0_T         0      NaN  242020000  242000000.0  200000000   
25271  242030209_0_T         0      NaN  242030209  242030200.0  242030000   
25272  243000000_0_T         0      NaN  243000000  200000000.0          0   
25273  243010000_0_T         0      NaN  243010000  243000000.0  200000000   
25274  243010101_0_T         0      NaN  243010101  243010100.0  243010000   
25275  243010107_0_T         0      NaN  243010107  243010100.0  243010000   
25276  243010300_0_T         0      NaN  243010300  243010000.0  243000000   
25277  243010404_0_T         0      NaN  243010404  243010400.0  243010000   
25278  243030103_0_T         0      NaN  243030103  243030100.0  243030000   
25279  243030200_0_T         0      NaN  243030200  243030000.0  243000000   
25280  243030300_0_T         0      NaN  243030300  243030000.0  243000000   
25281  244010104_0_T         0      NaN  244010104  244010100.0  244010000   
25282  244010700_0_T         0      NaN  244010700  244010000.0  244000000   
25283  244020102_0_T         0      NaN  244020102  244020100.0  244020000   
25284  244020104_0_T         0      NaN  244020104  244020100.0  244020000   
25285  244020202_0_T         0      NaN  244020202  244020200.0  244020000   
25286  244020400_0_T         0      NaN  244020400  244020000.0  244000000   
25287  244020500_0_T         0      NaN  244020500  244020000.0  244000000   
25288  244020506_0_T         0      NaN

#### # 확인 : 결국 0인 애들이 중복된거였어 : 얘네들이 1124개( 5830+1124 = 6954개 )


---

### # cate_deli 데이터를 살펴보자

In [44]:
cate_deli.head()

_id  dcate_cd   dcate_nm        id1          id2  \
3599  231010401_4100128_T   4100128   유아용 식탁의자  231010401  231010400.0   
3600  231010401_4100132_T   4100132   침대 가드/범퍼  231010401  231010400.0   
3601  231010101_4100134_T   4100134  가구 손잡이/부품  231010101  231010100.0   
3602  231010201_4100141_T   4100141    장식장/진열장  231010201  231010200.0   
3603  231010602_4100147_T   4100147   스프링 매트리스  231010602  231010600.0   

            id3        id4        id5  lcate_cd    lcate_nm     ...      \
3599  231010000  231000000  200000000   1100001  가구/침구/인테리어     ...       
3600  231010000  231000000  200000000   1100001  가구/침구/인테리어     ...       
3601  231010000  231000000  200000000   1100001  가구/침구/인테리어     ...       
3602  231010000  231000000  200000000   1100001  가구/침구/인테리어     ...       
3603  231010000  231000000  200000000   1100001  가구/침구/인테리어     ...       

       mcate_nm       name1 name2   name3    name4  name5   scate_cd  \
3599  가구/DIY/시공    유아 어린이가구  일반가구  가구/DIY  가구/홈/데코  배송상품군  3100020.0   
3600  가구/DIY/시공    유아 어린이가구  일반가구  가구/DIY  가구/홈/데코  배송상품군  3100020.0   
3601  가구/DIY/시공  DIY가구/가구리폼  가구리폼  가구/DIY  가구/홈/데코  배송상품군  3100022.0   
3602  가구/DIY/시공         거실장  거실가구  가구/DIY  가구/홈/데코  배송상품군  3100023.0   
3603  가구/DIY/시공        매트리스  침실가구  가구/DIY  가구/홈/데코  배송상품군  3100025.0   

          scate_nm type Unnamed: 20  
3599     유아/어린이 가구    T         NaN  
3600     유아/어린이 가구    T         NaN  
3601  가구DIY 부자재/용품    T         NaN  
3602      거실장/거실가구    T         NaN  
3603          매트리스    T         NaN  

[5 rows x 21 columns]

#### # id1 으로 sorting

In [46]:
cate_deli.sort_values(by='id1')

_id  dcate_cd          dcate_nm        id1  \
9108         231000000_0_T         0               NaN  231000000   
19162        231010000_0_T         0               NaN  231010000   
4194         231010100_0_T         0               NaN  231010100   
13581  231010101_4100136_T   4100136   이외 가구DIY 부자재/용품  231010101   
18602  231010101_4100177_T   4100177        이외 시공/리모델링  231010101   
8508   231010101_4100213_T   4100213  중고/리퍼/전시/스크래치 가구  231010101   
23540  231010101_4100179_T   4100179             중문 시공  231010101   
18595  231010101_4100135_T   4100135         가구리폼/조립가구  231010101   
3601   231010101_4100134_T   4100134         가구 손잡이/부품  231010101   
8504   231010101_4100178_T   4100178         주방/빌트인 시공  231010101   
23539  231010101_4100176_T   4100176             욕실 시공  231010101   
14229        231010200_0_T         0               NaN  231010200   
13584  231010201_4100142_T   4100142          좌탁/원목테이블  231010201   
23537  231010201_4100139_T   4100139            소파 테이블  231010201   
13583  231010201_4100140_T   4100140           이외 거실가구  231010201   
3602   231010201_4100141_T   4100141           장식장/진열장  231010201   
13582  231010201_4100138_T   4100138       거실협탁/사이드테이블  231010201   
23536  231010201_4100137_T   4100137           거실장/TV장  231010201   
19163        231010202_0_T         0               NaN  231010202   
8503   231010203_4100175_T   4100175            패브릭 소파  231010203   
8501   231010203_4100171_T   4100171         빈백/1인용 소파  231010203   
18601  231010203_4100169_T   4100169         가죽/카우치 소파  231010203   
3613   231010203_4100173_T   4100173             원목 소파  231010203   
8502   231010203_4100172_T   4100172              소파베드  231010203   
13590  231010203_4100170_T   4100170             리클라이너  231010203   
13591  231010203_4100174_T   4100174             이외 소파  231010203   
18600  231010204_4100167_T   4100167       원목/철재 수납 선반  231010204   
3612   231010204_4100165_T   4100165           원목 공간박스  231010204   
3611   231010204_4100164_T   4100164               신발장  231010204   
14230        231010300_0_T         0               NaN  231010300   
...                    ...       ...               ...        ...   
25113  244030602_4105760_T   4105760               목걸이  244030602   
20124  244030603_4105761_T   4105761                반지  244030603   
20289        244030604_0_T         0               NaN  244030604   
15355        244030605_0_T         0               NaN  244030605   
25294        244030606_0_T         0               NaN  244030606   
10062  244030607_4105762_T   4105762                발찌  244030607   
15193  244030607_4105764_T   4105764                팔찌  244030607   
5323         244030700_0_T         0               NaN  244030700   
15356        244030701_0_T         0               NaN  244030701   
20290        244030702_0_T         0               NaN  244030702   
15357        244030800_0_T         0               NaN  244030800   
25295        244030801_0_T         0               NaN  244030801   
5324         244030802_0_T         0               NaN  244030802   
10243        244030803_0_T         0               NaN  244030803   
20291        244030804_0_T         0               NaN  244030804   
10244        244030805_0_T         0               NaN  244030805   
15358        244030806_0_T         0               NaN  244030806   
10245        244030900_0_T         0               NaN  244030900   
10246        244030901_0_T         0               NaN  244030901   
10063  244030902_4105765_T   4105765         헤어악세서리/모자  244030902   
25296        245000000_0_T         0               NaN  245000000   
5325         245010000_0_T         0               NaN  245010000   
20292        245010100_0_T         0               NaN  245010100   
10247        245010101_0_T         0               NaN  245010101   
20293        245020000_0_T         0               NaN  245020000   
10248        245020100_0_T         0               NaN  245020100   
25297        24502010

#### # dcate_cd 으로 sorting

In [45]:
cate_deli.sort_values(by='dcate_cd')

_id  dcate_cd       dcate_nm        id1          id2  \
25298        245030000_0_T         0            NaN  245030000  245000000.0   
14239        232010207_0_T         0            NaN  232010207  232010200.0   
14240        232020104_0_T         0            NaN  232020104  232020100.0   
14241        232020105_0_T         0            NaN  232020105  232020100.0   
14242        232030101_0_T         0            NaN  232030101  232030100.0   
14243        232030102_0_T         0            NaN  232030102  232030100.0   
14244        232030202_0_T         0            NaN  232030202  232030200.0   
14238        232010100_0_T         0            NaN  232010100  232010000.0   
14245        232030300_0_T         0            NaN  232030300  232030000.0   
14247        232030400_0_T         0            NaN  232030400  232030000.0   
14248        232030601_0_T         0            NaN  232030601  232030600.0   
14249        232030602_0_T         0            NaN  232030602  232030600.0   
14250        232030701_0_T         0            NaN  232030701  232030700.0   
14251        232030800_0_T         0            NaN  232030800  232030000.0   
14252        232040000_0_T         0            NaN  232040000  232000000.0   
14246        232030306_0_T         0            NaN  232030306  232030300.0   
14237        231050100_0_T         0            NaN  231050100  231050000.0   
14236        231040100_0_T         0            NaN  231040100  231040000.0   
14235        231030101_0_T         0            NaN  231030101  231030100.0   
10241        244030400_0_T         0            NaN  244030400  244030000.0   
10242        244030501_0_T         0            NaN  244030501  244030500.0   
10243        244030803_0_T         0            NaN  244030803  244030800.0   
10244        244030805_0_T         0            NaN  244030805  244030800.0   
10245        244030900_0_T         0            NaN  244030900  244030000.0   
10246        244030901_0_T         0            NaN  244030901  244030900.0   
10247        245010101_0_T         0            NaN  245010101  245010100.0   
10248        245020100_0_T         0            NaN  245020100  245020000.0   
10249        245030101_0_T         0            NaN  245030101  245030100.0   
14229        231010200_0_T         0            NaN  231010200  231010000.0   
14230        231010300_0_T         0            NaN  231010300  231010000.0   
...                    ...       ...            ...        ...          ...   
10077  243020602_4105855_T   4105855          섹시 속옷  243020602  243020600.0   
20149  243020502_4105856_T   4105856         주니어 브라  243020502  243020500.0   
25131  243020602_4105857_T   4105857   이외 섹시 속옷/란제리  243020602  243020600.0   
25132  243020602_4105858_T   4105858         커플 란제리  243020602  243020600.0   
10078  243020502_4105859_T   4105859       누디/누드 브라  243020502  243020500.0   
20150  243020502_4105860_T   4105860  노와이어/심리스 브라세트  243020502  243020500.0   
5165   243020602_4105861_T   4105861    노와이어/심리스 브라  243020602  243020600.0   
20151  243020502_4105862_T   4105862       브라/팬티 세트  243020502  243020500.0   
20152  243020502_4105863_T   4105863      주니어 브라 세트  243020502  243020500.0   
25133  243020602_4105864_T   4105864       누디/누드 브라  243020602  243020600.0   
10079  243020602_4105865_T   4105865        브라 끈/후크  243020602  243020600.0   
25134  243020602_4105866_T   4105866         스포츠 브라  243020602  243020600.0   
5166   243020502_4105867_T   4105867      누디/노라인 팬티  243020502  243020500.0   
5167   243020502_4105868_T   4105868          사각 팬티  243020502  243020500.0   
25135  243020602_4105869_T   4105869      일반/볼륨업 브라  243020602  243020600.0   
5168   243020502_4105870_T   4105870          삼각 팬티  243020502  243020500.0   
15215  243020502_4105871_T   4105871        섹시/T 팬티  243020502  243020500.0   
25136  243020602_4105872_T   4105872         주니어 브라  243020602  243020600.0   
5169   243020502_4105873_T   4105873          위생 팬티  243020502  243020500

---

#### # cate_deli 데이터 안에서, unique한 id1으로 'dcate_cd'를 분류해보자.

In [81]:
cate_deli.id1.nunique()

2109

In [82]:
id_c_ = {}
nm_c_ = {}

for i in range(cate_deli.id1.nunique()):
    id_c_[i] = cate_[cate_.id1 == cate_deli.id1.unique()[i]].dcate_cd.values # category id 매핑정보를 얻는다
    nm_c_[i] = cate_[cate_.id1 == cate_deli.id1.unique()[i]].dcate_nm.values # category name 매핑정보를 얻는다


In [83]:
len(id_c_)

2109

In [84]:
ss = 0
for i in range(len(id_c_)):
    ss += len(id_c_[i])

In [85]:
ss

6954

In [86]:
cate_deli.shape

(6954, 21)

In [88]:
id_c_

{0: array([4100128, 4100132, 4100202, 4100129, 4100126, 4100203, 4100127,
        4100130, 4100131, 4100199, 4100200, 4100201]),
 1: array([4100134, 4100178, 4100213, 4100136, 4100135, 4100177, 4100176,
        4100179]),
 2: array([4100141, 4100138, 4100140, 4100142, 4100137, 4100139]),
 3: array([4100147, 4100148, 4100146, 4100149]),
 4: array([4100152, 4100153, 4100154, 4100151, 4100150, 4100155]),
 5: array([4100159, 4100161, 4100158, 4100160, 4100156, 4100157]),
 6: array([4100162, 4100163, 4100166, 4100168]),
 7: array([4100164, 4100165, 4100167]),
 8: array([4100173, 4100171, 4100172, 4100175, 4100170, 4100174, 4100169]),
 9: array([4100180, 4100186, 4100182, 4100183, 4100184, 4100181, 4100185]),
 10: array([4100193, 4100195, 4100198, 4100194, 4100196, 4100192, 4100197]),
 11: array([4100205, 4100207, 4100208, 4100204, 4100206]),
 12: array([4100209, 4100210, 4100211, 4100212]),
 13: array([4100225, 4100227, 4100224, 4100226, 4100228]),
 14: array([4100232, 4100236, 4100233, 410

In [89]:
nm_c_

{0: array(['유아용 식탁의자', '침대 가드/범퍼', '유아의자/유아소파', '유아용 옷걸이', '기저귀 정리함',
        '이외 유아/어린이 가구', '옷장/서랍장', '유아용 이동식행거', '유아침대/범퍼침대', '벙커침대/2층침대',
        '아기 책상/테이블', '어린이 책상/공부상'], dtype=object),
 1: array(['가구 손잡이/부품', '주방/빌트인 시공', '중고/리퍼/전시/스크래치 가구', '이외 가구DIY 부자재/용품',
        '가구리폼/조립가구', '이외 시공/리모델링', '욕실 시공', '중문 시공'], dtype=object),
 2: array(['장식장/진열장', '거실협탁/사이드테이블', '이외 거실가구', '좌탁/원목테이블', '거실장/TV장',
        '소파 테이블'], dtype=object),
 3: array(['스프링 매트리스', '이외 매트리스', '라텍스/메모리폼 매트리스', '접이식 매트리스'], dtype=object),
 4: array(['6인 이상 식탁', '밥상/다과상/교자상', '아일랜드 식탁', '4인용 식탁', '2인용 식탁',
        '이외 밥상/식탁'], dtype=object),
 5: array(['친환경 벽지', '합지 벽지', '이외 벽지', '포인트 벽지', '도배풀/도배용품', '보온/단열 벽지'],
       dtype=object),
 6: array(['라탄 서랍장', '목재/원목 서랍장', '원목 협탁', '철재/스틸 서랍장'], dtype=object),
 7: array(['신발장', '원목 공간박스', '원목/철재 수납 선반'], dtype=object),
 8: array(['원목 소파', '빈백/1인용 소파', '소파베드', '패브릭 소파', '리클라이너', '이외 소파',
        '가죽/카우치 소파'], dtype=object),
 9: array(['띠벽지/포인트/아이방 스티커', '현관문 시트지'

---

## # 여기서부턴, df2_re 를 이용!
# # <u>표준 카테고리(라이브딜) 하나에, 몇개의 물리카테고리(배송2.0)가 mapping되었나?<u/>

In [38]:
id_c = {}
nm_c = {}

for i in range(df2_re.shape[0]):
    id_c[i] = cate_[cate_.id1 == df2_re.dcateCd[i]].dcate_cd.values # category id 매핑정보를 얻는다
    nm_c[i] = cate_[cate_.id1 == df2_re.dcateCd[i]].dcate_nm.values # category name 매핑정보를 얻는다


In [39]:
id_c

{0: array([4101094, 4101092, 4105313, 4101095, 4101093, 4105238]),
 1: array([4101252, 4101253, 4101254, 4101257, 4101256, 4101255]),
 2: array([4105346, 4100975]),
 3: array([4105291, 4101073]),
 4: array([4101227, 4101228, 4101229, 4101231, 4101226, 4101230, 4101272,
        4101232]),
 5: array([4100938, 4105312]),
 6: array([4105738]),
 7: array([4105298, 4101083, 4105456]),
 8: array([4100983, 4105366]),
 9: array([4105324, 4100946]),
 10: array([4101103, 4101104]),
 11: array([4105310, 4100935]),
 12: array([4101132, 4105502, 4105504, 4105590, 4101135, 4105500, 4105506,
        4105589, 4105591, 4105592, 4105498, 4105501, 4105585, 4105593]),
 13: array([4101263, 4101258, 4101261, 4101262, 4101259, 4101260]),
 14: array([4100979, 4105353]),
 15: array([4105488, 4101122]),
 16: array([4101234, 4101235, 4101236, 4101233]),
 17: array([4105471, 4105473, 4105581, 4105584, 4105475, 4105478, 4105481,
        4105518, 4101113, 4105579, 4105480, 4105580, 4105595, 4101117,
        4105476,

In [40]:
nm_c

{0: array(['이외 유아/어린이 DVD', '동요집', '유아동DVD', '해외/영어 DVD', '애니메이션/교육만화',
        '유아동DVD'], dtype=object),
 1: array(['니콘', '삼성', '소니', '후지/파나소닉/올림푸스', '캐논', '이외 미러리스 렌즈'], dtype=object),
 2: array(['공포/추리/퇴마', '공포/추리/퇴마'], dtype=object),
 3: array(['초점책/헝겊책', '초점책/헝겊책'], dtype=object),
 4: array(['삼양옵틱스', '소니', '시그마', '캐논', '니콘', '이외 DSLR 렌즈', '렌즈', '탐론'],
       dtype=object),
 5: array(['액션/어드벤쳐/SF', '액션/어드벤쳐/SF'], dtype=object),
 6: array(['배송상품권'], dtype=object),
 7: array(['유아동 문화/예술/인물', '위인전/역사 동화', '유아동 문화/예술/인물'], dtype=object),
 8: array(['고전 명작/동화', '고전 명작/동화'], dtype=object),
 9: array(['재즈/클래식/뉴에이지', '재즈/클래식/뉴에이지'], dtype=object),
 10: array(['렌탈 전집/도서', '중고 전집/도서'], dtype=object),
 11: array(['뮤직비디오/콘서트', '뮤직비디오/콘서트'], dtype=object),
 12: array(['이외 어학/언어/사전', '이외 어학/언어/사전', '전자사전', '영어 시험 대비', '토익/토플/텝스',
        '영어 시험 대비', '한자', '번역', '영어로 씌어진 한국서적', '전자사전', '번역',
        '영어로 씌어진 한국서적', '경제/과학/문화사전', '한자'], dtype=object),
 13: array(['후지/파나소닉', '삼성', '이외 미러리스 카메라', '캐

## # 매핑 정보를 저장!

In [48]:
pd.DataFrame(dict( [ (k, pd.Series(v)) for k,v in nm_c.items() ] )).transpose().to_excel('cate_name.xlsx')

In [41]:
pd.DataFrame(dict( [ (k, pd.Series(v)) for k,v in id_c.items() ] )).transpose().to_excel('cate_number.xlsx')

---

### # 이건 실제 매핑 정보를 얻기 위해서, 검색해 얻은 스킬!

In [36]:
import numpy as np

In [37]:
d = dict( A = np.array([1,2]), B = np.array([1,2,3,4]) )

In [38]:
d

{'A': array([1, 2]), 'B': array([1, 2, 3, 4])}

In [43]:
d.items()

dict_items([('A', array([1, 2])), ('B', array([1, 2, 3, 4]))])

In [ ]:
d.

In [41]:
[ (k, pd.Series(v)) for k,v in d.items() ]

[('A', 0    1
  1    2
  dtype: int64), ('B', 0    1
  1    2
  2    3
  3    4
  dtype: int64)]

In [45]:
pd.DataFrame(dict( [ (k, pd.Series(v)) for k,v in d.items() ] )).transpose()

0    1    2    3
A  1.0  2.0  NaN  NaN
B  1.0  2.0  3.0  4.0

---

---

## # 매핑 정보들을 확인해보자

In [80]:
nm_c[1219][0]

nan

In [86]:
nm_c[0][0]

'이외 유아/어린이 DVD'

### # 아무것도 mapping 된것이 없다면, array에 nan값이 들어가 있을텐데,  nan의 type은 float이네!

In [85]:
type(nm_c[0][0]) == float

False

In [84]:
type(nm_c[1219][0]) == float

True

### # 이를 이용하여, mapping-x 카테고리의 정보들만 빼와보자! (인덱스 이용)

In [87]:
idx = []
for i in range(df2_re.shape[0]):
    if type(nm_c[i][0]) == float :
        idx.append(i)

In [88]:
idx

[808,
 809,
 810,
 811,
 812,
 813,
 814,
 815,
 816,
 817,
 818,
 819,
 820,
 821,
 822,
 823,
 825,
 826,
 827,
 828,
 829,
 830,
 831,
 832,
 833,
 834,
 835,
 836,
 837,
 838,
 839,
 840,
 841,
 842,
 843,
 844,
 845,
 846,
 847,
 848,
 849,
 850,
 851,
 852,
 854,
 855,
 856,
 857,
 858,
 859,
 860,
 861,
 862,
 863,
 864,
 865,
 866,
 867,
 868,
 869,
 870,
 871,
 872,
 873,
 874,
 875,
 876,
 877,
 878,
 879,
 880,
 881,
 882,
 883,
 884,
 885,
 886,
 887,
 888,
 889,
 890,
 891,
 892,
 893,
 894,
 895,
 896,
 897,
 898,
 899,
 900,
 901,
 902,
 903,
 904,
 905,
 906,
 907,
 908,
 909,
 910,
 911,
 912,
 913,
 914,
 915,
 916,
 917,
 918,
 919,
 920,
 921,
 922,
 923,
 924,
 925,
 926,
 927,
 928,
 929,
 930,
 931,
 932,
 933,
 934,
 935,
 936,
 937,
 938,
 939,
 940,
 941,
 942,
 943,
 944,
 945,
 946,
 947,
 948,
 949,
 950,
 951,
 952,
 953,
 954,
 955,
 956,
 957,
 958,
 960,
 961,
 962,
 963,
 964,
 965,
 966,
 967,
 969,
 970,
 971,
 972,
 973,
 974,
 975,
 976,
 977,
 978

In [89]:
len(idx)

594

---

## # category 매핑 안된 애들! - 그냥 자기 그대로의 표준카테고리!

In [106]:
df2_re.iloc[idx] # 얘네가 변동 없는 애들인데... 배송상품군이긴 한데?

Unnamed: 0  index    dcateCd  rcp_diff     rcp_x     rcp_y  \
808          808   1180  239030601       0.0  0.056956  0.056956   
809          809   1472  244020203       0.0  0.108015  0.108015   
810          810   1470  244020201       0.0  0.041667  0.041667   
811          811   1471  244020202       0.0  0.114190  0.114190   
812          812   1179  239030503       0.0  0.059943  0.059943   
813          813   1181  239030602       0.0  0.066038  0.066038   
814          814   1523  244030508       0.0  0.074253  0.074253   
815          815   1490  244020605       0.0  0.100699  0.100699   
816          816   1285  241010104       0.0  0.333916  0.333916   
817          817   1336  242010101       0.0  0.192039  0.192039   
818          818   1469  244020104       0.0  0.073517  0.073517   
819          819   1178  239030502       0.0  0.051625  0.051625   
820          820   1330  241030501       0.0  0.187414  0.187414   
821          821   1329  241030404       0.0  0.089233  0.089233   
822          822   1467  244020102       0.0  0.121603  0.121603   
823          823   1185  239030606       0.0  0.072249  0.072249   
825          825   1187  239030701       0.0  0.182715  0.182715   
826          826   1466  244020101       0.0  0.049602  0.049602   
827          827   1191  239030801       0.0  0.054833  0.054833   
828          828   1468  244020103       0.0  0.081482  0.081482   
829          829   1175  239030403       0.0  0.062589  0.062589   
830          830   1473  244020204       0.0  0.009488  0.009488   
831          831   1474  244020205       0.0  0.082381  0.082381   
832          832   1477  244020401       0.0  0.159857  0.159857   
833          833   1153  239021207       0.0  0.027027  0.027027   
834          834   1158  239030103       0.0  0.042130  0.042130   
835          835   1159  239030104       0.0  0.048611  0.048611   
836          836   1363  242030204       0.0  0.111111  0.111111   
837          837   1476  244020301       0.0  0.098725  0.098725   
838          838   1165  239030205       0.0  0.069767  0.069767   
...          ...    ...        ...       ...       ...       ...   
1384        1384    719  235040106       0.0  0.270644  0.270644   
1385        1385    720  235040107       0.0  0.187900  0.187900   
1386        1386    109  232020105       0.0  0.049020  0.049020   
1387        1387    721  235040108       0.0  0.064302  0.064302   
1388        1388    722  235040110       0.0  0.328647  0.328647   
1389        1389    723  235040111       0.0  0.099027  0.099027   
1390        1390    724  235040201       0.0  0.081194  0.081194   
1391        1391    725  235040301       0.0  0.071529  0.071529   
1392        1392    726  235040302       0.0  0.260870  0.260870   
1393        1393    708  235030108       0.0  0.000000  0.000000   
1394        1394    709  235030110       0.0  0.282720  0.282720   
1395        1395    696  235020201       0.0  0.153631  0.153631   
1396        1396    700  235020405       0.0  0.119008  0.119008   
1397        1397    690  235020104       0.0  0.160246  0.160246   
1398        1398    992  238050302       0.0  0.059543  0.059543   
1399        1399    691  235020106       0.0  0.145267  0.145267   
1400        1400    692  235020107       0.0  0.114169  0.114169   
1401        1401    693  235020108       0.0  0.059322  0.059322   
1402        1402    694  235020110       0.0  0.172426  0.172426   
1403        1403    695  235020111       0.0  0.077844  0.077844   
1404        1404    697  235020304       0.0  0.107793  0.107793   
1405        1405    698  235020401       0.0  0.110158  0.110158   
1406        1406    699  235020403       0.0  0.069554  0.069554   
1407        1407    936  238020102       0.0  0.006993  0.006993   
1408        1408    701  235030101       0.0  0.066351  0.066351   
1409        1409    702  235030102       0.0  0.105783  0.105783   
1410        1410    703  235030103       0.0  0.283080  0.2830

#### # 저장~

In [108]:
df2_re.iloc[idx].to_excel('df2_re_nochange.xlsx')

---

In [14]:
cate_.id1.nunique() # 표준 카테고리의 갯수

4199

In [15]:
cate_[cate_.id1<200000000].shape # 

(1773, 21)

In [16]:
cate_[(200000000 <= cate_.id1) & (cate_.id1 < 300000000)].id1.nunique()

2109

In [17]:
cate_[(200000000 <= cate_.id1) & (cate_.id1 < 300000000)].id1.unique()

array([231010401, 231010101, 231010201, ..., 245000000, 245020101,
       245030000])

---

In [109]:
rcp_table = pd.read_excel('(기준)김태준차장님_RCP_수정_3 복사본.xlsx')

In [110]:
rcp_table

구분    dcateCd          dcateNm    scateCd       scateNm    mcateCd  \
0     배송2.0    4104228             ASUS    3100745      2in1 노트북    2100132   
1     배송2.0    4104229             DELL    3100745      2in1 노트북    2100132   
2     배송2.0    4104230               HP    3100745      2in1 노트북    2100132   
3     배송2.0    4104231           Lenovo    3100745      2in1 노트북    2100132   
4     배송2.0    4104232               LG    3100745      2in1 노트북    2100132   
5     배송2.0    4104233        Microsoft    3100745      2in1 노트북    2100132   
6     배송2.0    4104234           이외 브랜드    3100745      2in1 노트북    2100132   
7     배송2.0    4104235               삼성    3100745      2in1 노트북    2100132   
8     배송2.0    4104423           3D 프린터    3100790     3D/라벨 프린터    2100141   
9     배송2.0    4104424           라벨 프린터    3100790     3D/라벨 프린터    2100141   
10    배송2.0    4104425       라벨 프린터 테이프    3100790     3D/라벨 프린터    2100141   
11     라이브딜  231010101       DIY가구/가구리폼  231010100          가구리폼  231010000   
12    배송2.0    4100134        가구 손잡이/부품    3100022  가구DIY 부자재/용품    2100006   
13    배송2.0    4100135        가구리폼/조립가구    3100022  가구DIY 부자재/용품    2100006   
14    배송2.0    4100136  이외 가구DIY 부자재/용품    3100022  가구DIY 부자재/용품    2100006   
15    배송2.0    4105739              가디건    3101053           가디건    2100202   
16    배송2.0    4104622          붙임머리 가발    3100827            가발    2100150   
17    배송2.0    4104623           앞머리 가발    3100827            가발    2100150   
18    배송2.0    4104624           전체/통가발    3100827            가발    2100150   
19    배송2.0    4101864      가발 테이프/관리용품    3100356            가발    2100060   
20    배송2.0    4101866          붙임머리 가발    3100356            가발    2100060   
21    배송2.0    4101867           앞머리 가발    3100356            가발    2100060   
22    배송2.0    4101868           전체/통가발    3100356            가발    2100060   
23     라이브딜  244030101             남성가방  244030100            가방  244030000   
24    배송2.0    4101565    강아지 유기농/수제 간식    3100299    강아지 간식/영양제    2100046   
25     라이브딜  244030103         여행용가방/소품  244030100            가방  244030000   
26     라이브딜  244030104            캐주얼가방  244030100            가방  244030000   
27     라이브딜  244020101             남성가방  244020100            가방  244020000   
28     라이브딜  244020102             여성가방  244020100            가방  244020000   
29     라이브딜  244020103         여행용가방/소품  244020100            가방  244020000   
...     ...        ...              ...        ...           ...        ...   
6952  배송2.0    4104365         데스크탑 RAM    3100776           RAM    2100138   
6953  배송2.0    4104366           서버 RAM    3100776           RAM    2100138   
6954  배송2.0    4101374             16GB    3100262     SD/SDHC카드    2100030   
6955  배송2.0    4101375             32GB    3100262     SD/SDHC카드    2100030   
6956  배송2.0    4101376          64GB 이상    3100262     SD/SDHC카드    2100030   
6957  배송2.0    4101377           8GB 이하    3100262     SD/SDHC카드    2100030   
6958  배송2.0    4104307              도시바    3100762           SSD    2100136   
6959  배송2.0    4104308               삼성    3100762           SSD    2100136   
6960  배송2.0    4104309             샌디스크    3100762           SSD    2100136   
6961  배송2.0    4104310           이외 브랜드    3100762           SSD    2100136   
6962  배송2.0    4104311             트랜샌드    3100762           SSD    2100136   
6963  배송2.0    4104295               LG    3100758        TV 모니터    2100135   
6964  배송2.0    4104296               삼성    3100758        TV 모니터    2100135   
6965  배송2.0    4104297           이외 브랜드    3100758        TV 모니터    2100135   
6966  배송2.0    4100469    TV 브라켓/스탠드 용품    3100088       TV 주변기기    2100010   
6967  배송2.0    4100470           TV 케이블    3100088       TV 주변기기    2100010   
6968  배송2.0    4100471              리모컨    3100088       TV 주변기기    2100010   
6969  배송2.0    4100472          안테나/증폭기    3100088       TV 주변기기    2100010   
6970  배송2.0    4100474       이외 TV 주변기기    3100088       TV

In [112]:
rcp_table[['dcateCd', 'dcateNm']]

dcateCd          dcateNm
0       4104228             ASUS
1       4104229             DELL
2       4104230               HP
3       4104231           Lenovo
4       4104232               LG
5       4104233        Microsoft
6       4104234           이외 브랜드
7       4104235               삼성
8       4104423           3D 프린터
9       4104424           라벨 프린터
10      4104425       라벨 프린터 테이프
11    231010101       DIY가구/가구리폼
12      4100134        가구 손잡이/부품
13      4100135        가구리폼/조립가구
14      4100136  이외 가구DIY 부자재/용품
15      4105739              가디건
16      4104622          붙임머리 가발
17      4104623           앞머리 가발
18      4104624           전체/통가발
19      4101864      가발 테이프/관리용품
20      4101866          붙임머리 가발
21      4101867           앞머리 가발
22      4101868           전체/통가발
23    244030101             남성가방
24      4101565    강아지 유기농/수제 간식
25    244030103         여행용가방/소품
26    244030104            캐주얼가방
27    244020101             남성가방
28    244020102             여성가방
29    244020103         여행용가방/소품
...         ...              ...
6952    4104365         데스크탑 RAM
6953    4104366           서버 RAM
6954    4101374             16GB
6955    4101375             32GB
6956    4101376          64GB 이상
6957    4101377           8GB 이하
6958    4104307              도시바
6959    4104308               삼성
6960    4104309             샌디스크
6961    4104310           이외 브랜드
6962    4104311             트랜샌드
6963    4104295               LG
6964    4104296               삼성
6965    4104297           이외 브랜드
6966    4100469    TV 브라켓/스탠드 용품
6967    4100470           TV 케이블
6968    4100471              리모컨
6969    4100472          안테나/증폭기
6970    4100474       이외 TV 주변기기
6971    4100475            LG TV
6972    4100476            삼성 TV
6973    4100477          중소기업 TV
6974    4104312           USB 동글
6975    4104313          USB 라이트
6976    4104314           USB 허브
6977    4104315      이외 USB 액세서리
6978    4104316             16GB
6979    4104317             32GB
6980    4104318          64GB 이상
6981    4104319           8GB 이하

[6982 rows x 2 columns]

In [115]:
df2_re_merge = pd.merge(df2_re, rcp_table[['dcateCd', 'dcateNm']], how='left' )

In [117]:
df2_re_merge.shape

(1542, 9)

In [118]:
df2_re.shape # 음 맞게 되었네!

(1542, 8)

In [120]:
df2_re_merge.head()

Unnamed: 0  index    dcateCd  rcp_diff     rcp_x     rcp_y  \
0           0     98  232010112  0.548803  0.118132  0.666935   
1           1    568  233090606  0.432881  0.040000  0.472881   
2           2    124  232040102  0.371951  0.000000  0.371951   
3           3    146  232050207  0.349552  0.014085  0.363636   
4           4    564  233090602  0.340484  0.253012  0.593496   

   totalOrderUserCnt_x  totalOrderUserCnt_y  dcateNm  
0                728.0                 2486   유아동/교육  
1                 50.0                  590  미러리스 렌즈  
2                 24.0                  164    공포/추리  
3                 71.0                  330  초점책/헝겊책  
4                166.0                  984  DSLR 렌즈

In [124]:
df2_re_merge.dcateNm.isnull().sum()  # 233020133 얘는 없네?

1

In [126]:
df2_re_merge.tail()

Unnamed: 0  index    dcateCd  rcp_diff     rcp_x     rcp_y  \
1537        1537    292  232063701 -0.063636  0.200000  0.136364   
1538        1538    565  233090603 -0.089369  0.173077  0.083708   
1539        1539    576  233100106 -0.107467  0.261719  0.154252   
1540        1540     99  232010114 -0.137592  0.272727  0.135135   
1541        1541    416  233020133 -0.184599  0.250000  0.065401   

      totalOrderUserCnt_x  totalOrderUserCnt_y      dcateNm  
1537                  5.0                   66      패션/뷰티일반  
1538                 52.0                 1111  PREMIUM 카메라  
1539                512.0                 4551          RAM  
1540                 11.0                   37  클래식/뮤지컬/오페라  
1541                 20.0                  474          NaN

In [127]:
df2_re_merge.iloc[idx]

Unnamed: 0  index    dcateCd  rcp_diff     rcp_x     rcp_y  \
808          808   1180  239030601       0.0  0.056956  0.056956   
809          809   1472  244020203       0.0  0.108015  0.108015   
810          810   1470  244020201       0.0  0.041667  0.041667   
811          811   1471  244020202       0.0  0.114190  0.114190   
812          812   1179  239030503       0.0  0.059943  0.059943   
813          813   1181  239030602       0.0  0.066038  0.066038   
814          814   1523  244030508       0.0  0.074253  0.074253   
815          815   1490  244020605       0.0  0.100699  0.100699   
816          816   1285  241010104       0.0  0.333916  0.333916   
817          817   1336  242010101       0.0  0.192039  0.192039   
818          818   1469  244020104       0.0  0.073517  0.073517   
819          819   1178  239030502       0.0  0.051625  0.051625   
820          820   1330  241030501       0.0  0.187414  0.187414   
821          821   1329  241030404       0.0  0.089233  0.089233   
822          822   1467  244020102       0.0  0.121603  0.121603   
823          823   1185  239030606       0.0  0.072249  0.072249   
825          825   1187  239030701       0.0  0.182715  0.182715   
826          826   1466  244020101       0.0  0.049602  0.049602   
827          827   1191  239030801       0.0  0.054833  0.054833   
828          828   1468  244020103       0.0  0.081482  0.081482   
829          829   1175  239030403       0.0  0.062589  0.062589   
830          830   1473  244020204       0.0  0.009488  0.009488   
831          831   1474  244020205       0.0  0.082381  0.082381   
832          832   1477  244020401       0.0  0.159857  0.159857   
833          833   1153  239021207       0.0  0.027027  0.027027   
834          834   1158  239030103       0.0  0.042130  0.042130   
835          835   1159  239030104       0.0  0.048611  0.048611   
836          836   1363  242030204       0.0  0.111111  0.111111   
837          837   1476  244020301       0.0  0.098725  0.098725   
838          838   1165  239030205       0.0  0.069767  0.069767   
...          ...    ...        ...       ...       ...       ...   
1384        1384    719  235040106       0.0  0.270644  0.270644   
1385        1385    720  235040107       0.0  0.187900  0.187900   
1386        1386    109  232020105       0.0  0.049020  0.049020   
1387        1387    721  235040108       0.0  0.064302  0.064302   
1388        1388    722  235040110       0.0  0.328647  0.328647   
1389        1389    723  235040111       0.0  0.099027  0.099027   
1390        1390    724  235040201       0.0  0.081194  0.081194   
1391        1391    725  235040301       0.0  0.071529  0.071529   
1392        1392    726  235040302       0.0  0.260870  0.260870   
1393        1393    708  235030108       0.0  0.000000  0.000000   
1394        1394    709  235030110       0.0  0.282720  0.282720   
1395        1395    696  235020201       0.0  0.153631  0.153631   
1396        1396    700  235020405       0.0  0.119008  0.119008   
1397        1397    690  235020104       0.0  0.160246  0.160246   
1398        1398    992  238050302       0.0  0.059543  0.059543   
1399        1399    691  235020106       0.0  0.145267  0.145267   
1400        1400    692  235020107       0.0  0.114169  0.114169   
1401        1401    693  235020108       0.0  0.059322  0.059322   
1402        1402    694  235020110       0.0  0.172426  0.172426   
1403        1403    695  235020111       0.0  0.077844  0.077844   
1404        1404    697  235020304       0.0  0.107793  0.107793   
1405        1405    698  235020401       0.0  0.110158  0.110158   
1406        1406    699  235020403       0.0  0.069554  0.069554   
1407        1407    936  238020102       0.0  0.006993  0.006993   
1408        1408    701  235030101       0.0  0.066351  0.066351   
1409        1409    702  235030102       0.0  0.105783  0.105783   
1410        1410    703  235030103       0.0  0.283080  0.2830

In [129]:
cate_name = pd.read_excel('cate_name.xlsx')

In [130]:
cate_name.head()

0         1       2             3           4           5    6   \
0  이외 유아/어린이 DVD       동요집  유아동DVD     해외/영어 DVD  애니메이션/교육만화      유아동DVD  NaN   
1             니콘        삼성      소니  후지/파나소닉/올림푸스          캐논  이외 미러리스 렌즈  NaN   
2       공포/추리/퇴마  공포/추리/퇴마     NaN           NaN         NaN         NaN  NaN   
3        초점책/헝겊책   초점책/헝겊책     NaN           NaN         NaN         NaN  NaN   
4          삼양옵틱스        소니     시그마            캐논          니콘  이외 DSLR 렌즈   렌즈   

    7    8    9  ...    55   56   57   58   59   60   61   62   63   64  
0  NaN  NaN  NaN ...   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
1  NaN  NaN  NaN ...   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
2  NaN  NaN  NaN ...   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
3  NaN  NaN  NaN ...   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
4   탐론  NaN  NaN ...   NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  

[5 rows x 65 columns]

In [131]:
cate_name.shape

(1542, 65)

In [134]:
df2_re_merge_new = pd.concat([df2_re_merge, cate_name], axis=1)

In [136]:
df2_re_merge_new.to_excel('df2_re_merge_new.xlsx')

In [139]:
df2_re_merge_new.iloc[idx].head()  # no-change데이터

Unnamed: 0  index    dcateCd  rcp_diff     rcp_x     rcp_y  \
808         808   1180  239030601       0.0  0.056956  0.056956   
809         809   1472  244020203       0.0  0.108015  0.108015   
810         810   1470  244020201       0.0  0.041667  0.041667   
811         811   1471  244020202       0.0  0.114190  0.114190   
812         812   1179  239030503       0.0  0.059943  0.059943   

     totalOrderUserCnt_x  totalOrderUserCnt_y      dcateNm    0 ...    55  \
808              20665.0                20665         깔창/끈  NaN ...   NaN   
809               3481.0                 3481       남성캐주얼화  NaN ...   NaN   
810                 24.0                   24          기능화  NaN ...   NaN   
811               1191.0                 1191       남성정장구두  NaN ...   NaN   
812               7724.0                 7724  스포츠 방한용품 기타  NaN ...   NaN   

      56   57   58   59   60   61   62   63   64  
808  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
809  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
810  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
811  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  
812  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  

[5 rows x 74 columns]

In [138]:
df2_re_merge_new.iloc[idx].to_excel('df2_re_merge_new_nochange.xlsx')

---

---

# # 추가, <u>하나의 물리카테고리(배송2.0)는 표준 카테고리(라이브딜) 하나에 mapping되었나?<u/>

In [70]:
# 예를 들어, 다음과 같은 물리카테고리의 mapping 정보를 살펴보자.

cate_deli.dcate_cd.unique()[0]

4100128

In [69]:
# 이러한 표준 카테고리에 mapping 되었구나! 

cate_deli[cate_deli.dcate_cd == cate_deli.dcate_cd.unique()[0]].id1.values

array([231010401])

In [80]:
cate_deli.dcate_cd.nunique()

5831

In [71]:
# 그렇다면 전체적인 mapping 정보를 살펴보자.
id_1231 = {}

for i in range(cate_deli.dcate_cd.nunique()):
    id_1231[i] = cate_deli[cate_deli.dcate_cd == cate_deli.dcate_cd.unique()[i]].id1.values # category id 매핑정보를 얻는다


In [72]:
id_1231

{0: array([231010401]),
 1: array([231010401]),
 2: array([231010101]),
 3: array([231010201]),
 4: array([231010602]),
 5: array([231010602]),
 6: array([231010502]),
 7: array([231010502]),
 8: array([231050302]),
 9: array([231050302]),
 10: array([231010604]),
 11: array([231010604]),
 12: array([231010204]),
 13: array([231010204]),
 14: array([231010203]),
 15: array([231050303]),
 16: array([231010605]),
 17: array([231010401]),
 18: array([231010402]),
 19: array([231010402]),
 20: array([231010402]),
 21: array([231050301]),
 22: array([231010405]),
 23: array([231010405]),
 24: array([231010606]),
 25: array([231010606]),
 26: array([231050304]),
 27: array([231050101]),
 28: array([231050203]),
 29: array([231050201]),
 30: array([231050103]),
 31: array([231050202]),
 32: array([231060203]),
 33: array([231060203]),
 34: array([231060204]),
 35: array([231060202]),
 36: array([231060206]),
 37: array([231060206]),
 38: array([231060207]),
 39: array([231060301]),
 40: array

In [74]:
# 252index의 물리카테고리의 mapping 정보 살펴보면 
cate_deli.dcate_cd.unique()[252]

0

In [75]:
# 이렇게 많은 표준카테고리로 mapping 되어있는것을 살필 수 있다. - 하지만 이건 예외적인 경우다.
cate_deli[cate_deli.dcate_cd == cate_deli.dcate_cd.unique()[252]].id1.values

array([231010100, 231010603, 231020000, ..., 245000000, 245020101,
       245030000])

In [73]:
pd.DataFrame(dict( [ (k, pd.Series(v)) for k,v in id_1231.items() ] )).transpose().to_excel('cate_1231.xlsx')